mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoGetEntityMethodGenerator.java

```diff
@@ -93,8 +96,12 @@ public DaoGetEntityMethodGenerator(
     TypeElement entityElement = null;
     Transformation transformation = null;
     TypeMirror returnType = methodElement.getReturnType();
-    if (returnType.getKind() == TypeKind.DECLARED) {
-      Element element = ((DeclaredType) returnType).asElement();
+    if (couldBeEntity(returnType)) {
```

Comment:
I initially factored it into a separate method as was worried about missing a case and thought this may make it easier to communicate intent, but since it ended up only being needed in two places i'll change it back.